uamqp/aio/_cbs_async.py

```diff
@@ -0,0 +1,227 @@
+                         "CBS error occurred on connection %r.", self._connection._container_id)
+        elif self.state == CbsState.OPENING:
+            self.state = CbsState.OPEN if management_open_result == ManagementOpenResult.OK else CbsState.CLOSED
+            _LOGGER.info("CBS for connection %r completed opening with status: %r",
+                         self._connection._container_id, management_open_result)
+
+    async def _on_amqp_management_error(self):
+        if self.state == CbsState.CLOSED:
+            _LOGGER.info("Unexpected AMQP error in CLOSED state.")
+        elif self.state == CbsState.OPENING:
+            self.state = CbsState.ERROR
+            await self._mgmt_link.close()
+            _LOGGER.info("CBS for connection %r completed opening with status: %r",
+                         self._connection._container_id, ManagementOpenResult.ERROR)
+        elif self.state == CbsState.OPEN:
+            self.state = CbsState.ERROR
+            _LOGGER.info("CBS error occurred on connection %r.", self._connection._container_id)
+
+    async def _on_execute_operation_complete(
+            self,
+            execute_operation_result,
+            status_code,
+            status_description,
+            message,
+            error_condition=None
+    ):
+        _LOGGER.info("CBS Put token result (%r), status code: %s, status_description: %s.",
+                     execute_operation_result, status_code, status_description)
+        self._token_status_code = status_code
+        self._token_status_description = status_description
+
+        if execute_operation_result == ManagementExecuteOperationResult.OK:
+            self.auth_state = CbsAuthState.Ok
+        elif execute_operation_result == ManagementExecuteOperationResult.ERROR:
+            self.auth_state = CbsAuthState.Error
+            # put-token-message sending failure, rejected
+            self._token_status_code = 0
+            self._token_status_description = "Auth message has been rejected."
+        elif execute_operation_result == ManagementExecuteOperationResult.FAILED_BAD_STATUS:
+            self.auth_state = CbsAuthState.Error
+
+    def _update_status(self):
+        # TODO: this is duplication the sync version, considering mixin or inheritance
+        if self.state == CbsAuthState.Ok or self.state == CbsAuthState.RefreshRequired:
+            is_expired, is_refresh_required = self._check_expiration_and_refresh_status()
+            if is_expired:
+                self.state = CbsAuthState.Expired
+            elif is_refresh_required:
+                self.state = CbsAuthState.RefreshRequired
+        elif self.state == CbsAuthState.InProgress:
+            put_timeout = self._check_put_timeout_status()
+            if put_timeout:
+                self.state = CbsAuthState.Timeout
+
+    def _check_expiration_and_refresh_status(self):
+        # TODO: this is duplication the sync version, considering mixin or inheritance
```

Comment:
Or standalone function? Nothing is being modified on self.

Reply:
Same comment applies to the functions below.

Reply:
nice suggestion, I was wondering how to avoid duplicate code sync/async module, making them standalone function help resolved the issue.